packages/react-dropdown/src/Categories.js

```diff
@@ -78,6 +82,62 @@ const GroupTitle = styled.h3`
   padding: 5px;
 `;
 
+type CategorySelectionProps = {
+  categoryId: string | number,
+  enabled: boolean,
+  children: React.Node,
+  getItemProps: GetItemProps,
+  items: Array<$Shape<DropdownItemWithIndex>>,
+};
+
+export const createCategoryIndex = (categoryId: string | number): string => {
+  return `group_${categoryId}`;
+};
+
+export const isEntityHighlighted = (
+  enabled: boolean = false,
+  highlightedIndex: ?number | ?string,
+  id: ?number | ?string
```

Comment:
May we define `?number | ?string` as `type ID = number | string` and use `?ID` instead? It's gonna be more readable

Reply:
I named it as: `HighlightedIndex`

Reply:
So that means we'll have the following?

```
  highlightedIndex: ?HighlightedIndex,
  id: ?HighlightedIndex,
```